Engine/ResourceManagement/Resources/Material.cpp

```diff
@@ -184,20 +176,18 @@ std::string Material::GetMaterialTypeName(const MaterialType material_type)
 		case MaterialType::MATERIAL_LIQUID:
 			return "Liquid";
 
+		case MaterialType::MATERIAL_DISSOLVING:
+			return "Dissolving";
+
 		default:
 			return "";
 	}
 }
 
 void Material::UpdateLiquidProperties()
 {
-	//TODO->change it to liquid maps and not hardcoded
-	//tiling_liquid_x_x += speed_tiling_x / 1000 * App->time->delta_time;
-	//tiling_liquid_x_y += speed_tiling_x / 1000 * App->time->delta_time;
-	tiling_liquid_y_x -= speed_tiling_y / 1000 * App->time->delta_time;
-	tiling_liquid_y_y -= speed_tiling_y / 1000 * App->time->delta_time;
-	tiling_liquid_x_x += speed_tiling_x / 1000 * App->time->delta_time;
-	tiling_liquid_x_y += speed_tiling_y / 1000 * App->time->delta_time;
+	liquid_vertical_normals_tiling += float2(liquid_tiling_speed.y * App->time->delta_time * 0.001f);
+	liquid_horizontal_normals_tiling += float2(liquid_tiling_speed.x * App->time->delta_time * 0.001f);
```

Comment:
vertical has to sum and horizontal has to substract

Reply:
True that!

Reply:
or viceversa, otherwise, they go at the same direction and it does not give the effect of randomness that you want from water

Reply:
Dont worry, I ve forgot to change that when simplifing

Reply:
Fixed!